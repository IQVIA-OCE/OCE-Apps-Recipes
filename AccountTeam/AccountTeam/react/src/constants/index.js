import { Platform } from 'react-native';

export { SORT_OPTION_LIST, ACCESS_LEVEL_PICKLIST } from './sortOptionsList';
export { NAMESPACE, LOCALIZATION_NAMESPACE } from './namespacePrefix';
export { LOADING_STATUS } from './loadingStatus';
export const isIphone = !Platform.isPad && Platform.OS !== 'web';
