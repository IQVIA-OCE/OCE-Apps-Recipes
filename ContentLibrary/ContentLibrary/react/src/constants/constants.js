export const OWNED_BY_ME = 'ownedByMe';
export const SHARED_WITH_ME = 'sharedWithMe';
export const NO_CONNECTION = 'No Connection';