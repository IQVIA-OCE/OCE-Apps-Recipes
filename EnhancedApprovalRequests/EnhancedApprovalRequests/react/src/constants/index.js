export { LOADING_STATUS } from './loadingStatus';
export { APPROVAL_REQUEST_STATUS } from './approvalRequestStatus';
export * from './sorting';
