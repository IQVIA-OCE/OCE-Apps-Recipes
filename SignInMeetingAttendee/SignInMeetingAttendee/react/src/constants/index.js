import { Platform } from 'react-native';

export { NAMESPACE, userId, profileId, organizationId } from './namespacePrefix';
export const isIphone = !Platform.isPad;