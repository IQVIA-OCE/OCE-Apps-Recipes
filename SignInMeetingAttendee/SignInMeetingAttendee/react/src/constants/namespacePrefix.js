import { environment } from 'oce-apps-bridges';

export const NAMESPACE = environment.namespace();

export const userId = environment.userID();

export const profileId = environment.profileId();

export const organizationId = environment.organizationId();
