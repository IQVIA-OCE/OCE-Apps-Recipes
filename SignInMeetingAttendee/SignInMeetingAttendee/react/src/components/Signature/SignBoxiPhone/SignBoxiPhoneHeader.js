import React, { useState, useEffect } from 'react';
import { View, StyleSheet } from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import { Checkbox, Paragraph, Select, themeGrey } from '@oce-apps/apollo-react-native';
import { updateMeal, updateMealOption } from '../../../stores/meeting'
import { NAMESPACE } from '../../../constants'


const SignBoxiPhoneHeader = () => {
    const [pickListOptions, setPickListOptions] = useState([]);
    const [selectedMealOption, setSelectedMealOption] = useState(null);
    const [isMeal, setIsMeal] = useState(false);
    const { meetingLayouts, meetingConfig } = useSelector((state) => state.meeting);
    useEffect(() => {
        if (meetingConfig[0] && meetingConfig[0][`${NAMESPACE}MealFieldType__c `] === 'Picklist') {
            const mealFieldType = meetingLayouts.fields.filter((layout) => layout.name === `${NAMESPACE}MealOption__c`);
            setPickListOptions(mealFieldType[0].picklistValues)
        }

    }, [meetingConfig]);
    const dispatch = useDispatch();
    const onIsMeal = () => {
        setIsMeal(!isMeal);
        dispatch(updateMeal(!isMeal))
    }
    const onChangeMealOption = (selected) => {
        setSelectedMealOption(selected);
        dispatch(updateMealOption(selected.value))
    }
    return (
        <View style={styles.mealRow}>
            <Paragraph style={styles.mealLabel}>{'Meal?'}</Paragraph>
            {meetingConfig[0] && meetingConfig[0][`${NAMESPACE}MealFieldType__c`] === 'Picklist'
                ? <View style={styles.mealPicker} testID={'Select'}>
                    <Select
                        options={pickListOptions}
                        value={selectedMealOption}
                        onChange={onChangeMealOption}
                        placeholder="Select Meal Option..."
                        style={styles.mealOptionSelect}
                        width={160}
                        testID={'meal-option'}
                    />
                </View>
                : <Checkbox onChange={onIsMeal} checked={isMeal} size='small' testID={'Checkbox'}/>
            }
        </View>
    )
};


export default SignBoxiPhoneHeader;

const styles = StyleSheet.create({
    mealRow: {
        flexDirection: 'row',
        marginTop: 15,
        alignItems: 'center',
        justifyContent: 'flex-end'
    },
    mealOptionSelect: {
        borderWidth: 1,
        borderColor: themeGrey[400],
        marginLeft: 10
    },
    mealLabel: {
        fontWeight: 'bold',
        fontSize: 16
    }
});
