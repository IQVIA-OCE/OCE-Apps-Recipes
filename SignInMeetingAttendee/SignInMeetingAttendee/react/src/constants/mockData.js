import { NAMESPACE } from '../constants';

import { localized } from 'oce-apps-bridges';

const meetingLevelConfigRecord = [{
    [`${NAMESPACE}MealFieldType__c`]: "Picklist"
}];
const meetingLevelConfigRecordForcheck = [{
    [`${NAMESPACE}MealFieldType__c`]: "Check"
}];
const meetingMemberLayout = {
    mergeable: false,
    recordTypeInfos: [
        {
            urls: {
                layout: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/layouts/0129D000000vAQNQA2`
            },
            master: false,
            defaultRecordTypeMapping: true,
            available: true,
            developerName: "Attendee",
            active: true,
            recordTypeId: "0129D000000vAQNQA2",
            name: "Attendee"
        },
        {
            urls: {
                layout: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/layouts/0129D000000vAQOQA2`
            },
            master: false,
            defaultRecordTypeMapping: false,
            available: true,
            developerName: "Colleague",
            active: true,
            recordTypeId: "0129D000000vAQOQA2",
            name: "Colleague"
        }
    ],
    customSetting: false,
    isInterface: false,
    custom: true,
    label: "Meeting Member",
    replicateable: true,
    supportedScopes: [
        {
            name: "everything",
            label: "All meeting members"
        }
    ],
    actionOverrides: [
        {
            isAvailableInTouch: false,
            formFactor: "LARGE",
            url: null,
            pageId: "0M09D0000009clxSAA",
            name: "View"
        }
    ],
    compactLayoutable: true,
    associateParentEntity: null,
    extendsInterfaces: null,
    updateable: true,
    sobjectDescribeOption: "FULL",
    extendedBy: null,
    retrieveable: true,
    hasSubtypes: false,
    listviewable: null,
    namedLayoutInfos: [],
    implementedBy: null,
    isSubtype: false,
    associateEntityType: null,
    keyPrefix: "a42",
    undeletable: true,
    name: `${NAMESPACE}MeetingMember__c`,
    lookupLayoutable: null,
    searchLayoutable: false,
    deepCloneable: false,
    layoutable: true,
    deprecatedAndHidden: false,
    mruEnabled: false,
    createable: true,
    triggerable: true,
    fields: [
        {
            cascadeDelete: false,
            mask: null,
            createable: true,
            picklistValues: [
                {
                    defaultValue: false,
                    label: "Breakfast",
                    active: true,
                    value: "Breakfast",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Lunch",
                    active: true,
                    value: "Lunch",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Snack",
                    active: true,
                    value: "Snack",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Dinner",
                    active: true,
                    value: "Dinner",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Lunch & Snack",
                    active: true,
                    value: "Lunch & Snack",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Did Not Consume",
                    active: true,
                    value: "Did Not Consume",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Not Applicable",
                    active: true,
                    value: "Not Applicable",
                    validFor: null
                }
            ],
            referenceTo: [],
            formulaTreatNullNumberAsZero: false,
            relationshipName: null,
            restrictedDelete: false,
            soapType: "xsd:string",
            idLookup: false,
            defaultValue: null,
            inlineHelpText: null,
            maskType: null,
            custom: true,
            autoNumber: false,
            defaultedOnCreate: false,
            aggregatable: true,
            controllerName: null,
            type: "picklist",
            dependentPicklist: false,
            referenceTargetField: null,
            htmlFormatted: false,
            namePointing: false,
            filterable: true,
            aiPredictionField: false,
            digits: 0,
            length: 255,
            calculated: false,
            deprecatedAndHidden: false,
            byteLength: 765,
            precision: 0,
            name: `${NAMESPACE}MealOption__c`,
            filteredLookupInfo: null,
            displayLocationInDecimal: false,
            scale: 0,
            groupable: true,
            defaultValueFormula: null,
            compoundFieldName: null,
            polymorphicForeignKey: false,
            nillable: true,
            encrypted: false,
            restrictedPicklist: false,
            highScaleNumber: false,
            sortable: true,
            nameField: false,
            permissionable: true,
            writeRequiresMasterRead: false,
            queryByDistance: false,
            unique: false,
            extraTypeInfo: null,
            externalId: false,
            searchPrefilterable: false,
            caseSensitive: false,
            relationshipOrder: null,
            updateable: true,
            calculatedFormula: null,
            label: "Meal Option"
        },
        {
            cascadeDelete: false,
            mask: null,
            createable: true,
            picklistValues: [],
            referenceTo: [],
            formulaTreatNullNumberAsZero: false,
            relationshipName: null,
            restrictedDelete: false,
            soapType: "xsd:boolean",
            idLookup: false,
            defaultValue: false,
            inlineHelpText: "Determines if the Meeting Member requires a meal at the meeting.",
            maskType: null,
            custom: true,
            autoNumber: false,
            defaultedOnCreate: true,
            aggregatable: false,
            controllerName: null,
            type: "boolean",
            dependentPicklist: false,
            referenceTargetField: null,
            htmlFormatted: false,
            namePointing: false,
            filterable: true,
            aiPredictionField: false,
            digits: 0,
            length: 0,
            calculated: false,
            deprecatedAndHidden: false,
            byteLength: 0,
            precision: 0,
            name: `${NAMESPACE}Meal__c`,
            filteredLookupInfo: null,
            displayLocationInDecimal: false,
            scale: 0,
            groupable: true,
            defaultValueFormula: null,
            compoundFieldName: null,
            polymorphicForeignKey: false,
            nillable: false,
            encrypted: false,
            restrictedPicklist: false,
            highScaleNumber: false,
            sortable: true,
            nameField: false,
            permissionable: true,
            writeRequiresMasterRead: false,
            queryByDistance: false,
            unique: false,
            extraTypeInfo: null,
            externalId: false,
            searchPrefilterable: false,
            caseSensitive: false,
            relationshipOrder: null,
            updateable: true,
            calculatedFormula: null,
            label: "Meal?"
        }
    ],
    deletable: true,
    activateable: false,
    searchable: false,
    childRelationships: [
        {
            junctionReferenceTo: [],
            deprecatedAndHidden: false,
            childSObject: "AttachedContentDocument",
            restrictedDelete: false,
            relationshipName: "AttachedContentDocuments",
            cascadeDelete: true,
            field: "LinkedEntityId",
            junctionIdListNames: []
        }
    ],
    urls: {
        layouts: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/layouts`,
        quickActions: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/quickActions`,
        sobject: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c`,
        uiNewRecord: `https://oce-pipeline-2--dcdm201002.my.salesforce.com/a42/e`,
        describe: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe`,
        compactLayouts: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/compactLayouts`,
        approvalLayouts: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/approvalLayouts`,
        uiDetailTemplate: "https://oce-pipeline-2--dcdm201002.my.salesforce.com/{ID}",
        rowTemplate: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/{ID}`,
        uiEditTemplate: "https://oce-pipeline-2--dcdm201002.my.salesforce.com/{ID}/e"
    },
    implementsInterfaces: null,
    feedEnabled: false,
    queryable: true,
    defaultImplementation: null,
    labelPlural: "Meeting Members",
    networkScopeFieldName: null
};


const meetingsInfo = [{
    CreatedBy: {
        Name: "Sales Representative PHX2"
    },
    CreatedById: "0056F00000B45efQAB",
    CreatedDate: "2021-08-24T18:35:54.000+0000",
    Id: "a470k0000010AMgAAM",
    Name: "Test1",
    [`${NAMESPACE}EndDateTime__c`]: "2021-08-31T18:35:00.000+0000",
    [`${NAMESPACE}Location__c`]: null,
    [`${NAMESPACE}StartDateTime__c`]: "2021-08-30T18:35:00.000+0000",
    [`${NAMESPACE}Status__c`]: "Draft",
    [`${NAMESPACE}Topic__c`]: null,
    RecordType: {
        Name: "Speaker Meeting"
    },
    RecordTypeId: "0120p0000012JBgAAM",
    uid: "a470k0000010AMgAAM",

}];


const meetingsMemebersInfo = [{
    Id: "a420k0000005MvUAAU",
    Name: "ASHOK KUMAR",
    [`${NAMESPACE}Address__c`]: null,
    [`${NAMESPACE}AttendanceStatus__c`]: "Attended-eSigned",
    [`${NAMESPACE}AttendeeType__c`]: null,
    [`${NAMESPACE}DiscrepancyReason__c`]: "Exceeded Monetary Caps Limit Warning",
    [`${NAMESPACE}MealOption__c`]: "Dinner",
    [`${NAMESPACE}SignatureDate__c`]: "2021-08-27T10:55:43.000+0000",
    [`${NAMESPACE}SpecialtyText__c`]: "Family medicine",
    [`${NAMESPACE}Status__c`]: "Invited",
    RecordTypeId: "0120p0000012JBOAA2",
    uid: "a420k0000005MvUAAU",
}, {
    Id: "a420k0000005Mv5AAE",
    Name: "Sales Representative PHX2",
    [`${NAMESPACE}Address__c`]: null,
    [`${NAMESPACE}AttendanceStatus__c`]: "Attended-eSigned",
    [`${NAMESPACE}AttendeeType__c`]: null,
    [`${NAMESPACE}DiscrepancyReason__c`]: null,
    [`${NAMESPACE}MealOption__c`]: null,
    [`${NAMESPACE}SignatureDate__c`]: null,
    [`${NAMESPACE}SpecialtyText__c`]: null,
    [`${NAMESPACE}Status__c`]: "Accepted",
    RecordTypeId: "0120p0000012JBPAA2",
    uid: "a420k0000005Mv5AAE",
}];

const meetingConfigData = [{
    [`${NAMESPACE}AttendStatusesForTopicCompliance__c`]: null,
    [`${NAMESPACE}EnableConsent__c`]: 1,
    [`${NAMESPACE}EnableWriteIns__c`]: 1,
    [`${NAMESPACE}IsEnabledSignInDateValidation__c`]: 1,
    [`${NAMESPACE}SignInAttendanceStatuses__c`]: null,
    [`${NAMESPACE}SignInInvitationStatuses__c`]: null,
    [`${NAMESPACE}ValidateDaysInPastForTopicForWalkIn__c`]: null,
    [`${NAMESPACE}ValidateDaysInPastForTopic__c`]: null,
    [`${NAMESPACE}ValidateRestrictedProductsForWalkIns__c`]: 'warn',

}];

const records = [{
    CreatedBy: {
        Name: "Sales Representative PHX2"
    },
    CreatedById: "0056F00000B45efQAB",
    CreatedDate: "2021-08-24T18:35:54.000+0000",
    Id: "a470k0000010AMgAAM",
    Name: "Test1",
    [`${NAMESPACE}EndDateTime__c`]: "2021-10-13T18:35:00.000+0000",
    [`${NAMESPACE}Location__c`]: 'Test Location',
    [`${NAMESPACE}StartDateTime__c`]: "2021-10-13T18:35:00.000+0000",
    [`${NAMESPACE}Status__c`]: "Draft",
    [`${NAMESPACE}Topic__c`]: null,
    ['RecordType.Name']: "Speaker Meeting",
    RecordTypeId: "0120p0000012JBgAAM",
    uid: "a470k0000010AMgAAM",
    [`${NAMESPACE}MeetingMember__r`]: {
        records: meetingsMemebersInfo
    }


}];

const singleRecordRow = {
    Id: "a421s000000CuyfAAC",
    Name: "Test A1",
    [`${NAMESPACE}Address__c`]: null,
    [`${NAMESPACE}AttendanceStatus__c`]: "Attended-eSigned",
    [`${NAMESPACE}AttendeeType__c`]: null,
    [`${NAMESPACE}DiscrepancyReason__c`]: null,
    [`${NAMESPACE}MealOption__c`]: "Dinner",
    [`${NAMESPACE}SignatureDate__c`]: "2021-09-15T10:33:45.421+05:30",
    [`${NAMESPACE}SpecialtyText__c`]: "Addiction medicine",
    [`${NAMESPACE}Status__c`]: "Invited",
    RecordTypeId: "0120k000000GlNkAAK",
    key: "a421s000000CuyfAAC",
    uid: "a421s000000CuyfAAC",

}

const meetingHighLightPanel = [{
    fieldItems: [{
        label: localized('Name', 'Meeting Name'),
        layoutComponents: [{
            type: "Field",
            value: "Name",
            details: {
                name: "Name",
                type: "string"
            }
        }]
    },
    {
        label: `${localized('Start', 'Start')}${localized('Date_time', 'Date/Time')}`,
        layoutComponents: [{
            type: `Field`,
            value: `${NAMESPACE}StartDateTime__c`,
            details: {
                name: `${NAMESPACE}StartDateTime__c`,
                type: "datetime"
            }
        }],
    },
    {
        label: `${localized('InputDateRange_EndDate_DefaultLabel', 'End')}${localized('Date_time', 'Date/Time')}`,
        layoutComponents: [{
            type: "Field",
            value: `${NAMESPACE}EndDateTime__c`,
            details: {
                name: `${NAMESPACE}EndDateTime__c`,
                type: "datetime"
            }
        }],
    },
    {
        label: `${localized('AttachmentCreatedByIdLabel', 'Created By')}`,
        layoutComponents: [
            {
                type: "Field",
                value: "CreatedById",
                details: {
                    name: "CreatedBy",
                    type: "reference",
                    relationshipName: 'CreatedBy'
                }
            },
            {
                type: "Separator",
                value: ", ",
            },
            {
                type: "Field",
                value: "CreatedDate",
                details: {
                    name: "CreatedDate",
                    type: "datetime"
                }
            },
        ],
    },
    {
        label: `${localized('Location', 'Location')}`,
        layoutComponents: [{
            type: "Field",
            value: `${NAMESPACE}Location__c`,
            details: {
                name: `${NAMESPACE}Location__c`,
                type: "string"
            }
        }],
    }]
}]

const meetingHighLight = [{
    actions: null,
    fieldItems: [{
        editableForNew: true,
        editableForUpdate: true,
        label: "Meeting Name",
        layoutComponents: [{
            displayLines: 1,
            tabOrder: 1,
            type: "Field",
            value: "Name",
            details: {
                label: "Meeting Name",
                length: 80,
                mask: null,
                maskType: null,
                name: "Name",
                type: "string",
                unique: false,
                updateable: true,
                writeRequiresMasterRead: false
            }
        }],
        placeholder: false,
        required: false,
    }, {
        editableForNew: true,
        editableForUpdate: true,
        label: "Start Date/Time",
        layoutComponents: [{
            displayLines: 1,
            tabOrder: 2,
            type: "Field",
            value: [`${NAMESPACE}StartDateTime__c`],
            details: {
                label: "Start Date/Time",
                length: 0,
                mask: null,
                maskType: null,
                name: [`${NAMESPACE}StartDateTime__c`],
                type: "datetime"
            }
        }],
        placeholder: false,
        required: false,
    }],
    id: "0AH0k000000CspjGAC",
    imageItems: null,
    label: "Summary",
    name: [`${NAMESPACE}Summary`],
    objectType: [`${NAMESPACE}Meeting__c`],
}]


export {
    records, meetingLevelConfigRecord, meetingMemberLayout,
    meetingsInfo, meetingsMemebersInfo, meetingConfigData,
    singleRecordRow, meetingHighLight, meetingHighLightPanel, meetingLevelConfigRecordForcheck
};

