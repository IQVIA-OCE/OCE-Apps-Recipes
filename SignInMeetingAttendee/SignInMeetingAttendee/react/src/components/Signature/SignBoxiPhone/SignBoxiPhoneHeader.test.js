import React from 'react';
import { NAMESPACE } from '../../../constants';
import SignBoxiPhoneHeader from './SignBoxiPhoneHeader';
import { fireEvent, render } from '@testing-library/react-native';
import { useSelector, useDispatch } from 'react-redux';
import { records, meetingLevelConfigRecord } from "../../../constants/mockData";
import { Checkbox, Select } from '@oce-apps/apollo-react-native';

jest.useFakeTimers();


jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

const mockUseRef = (obj) => () => Object.defineProperty({}, 'current', {
    get: () => obj,
    set: () => { }
})

describe('Signature Box Header for iPhone', () => {
    it('SignHeader Component with picklist selectbox for iPhone should render properly', async () => {
        useDispatch.mockReturnValue(() => jest.fn());
        useSelector.mockImplementation((cb) => cb({
            meeting: {
                meetingDetails: records,
                meetingConfig: meetingLevelConfigRecord,
                meetingLayouts: {
                    fields: [{
                        name: `${NAMESPACE}MealOption__c`,
                        label: 'Lunch',
                    },
                    {
                        name: `${NAMESPACE}Mealcheck__c`,
                        label: 'Dinner',
                    }]
                }
            }
        }));
        const { getByTestId, UNSAFE_getByType } = render(
            <SignBoxiPhoneHeader />
        );

        expect(getByTestId('Select')).toBeTruthy();
        UNSAFE_getByType(Select).props.onChange({ label: 'All', value: 'All' })
    })
    it('SignHeader Component with meal option checkbox for iPhone should render properly', async () => {
        const updatedArr = { ...meetingLevelConfigRecord, [`${NAMESPACE}MealFieldType__c`]: 'check' }
        useDispatch.mockReturnValue(() => jest.fn());
        useSelector.mockImplementation((cb) => cb({
            meeting: {
                meetingDetails: records,
                meetingConfig: [updatedArr],
            }
        }));
        const { getByTestId, UNSAFE_getByType } = render(
            <SignBoxiPhoneHeader />
        );
        UNSAFE_getByType(Checkbox).props.onChange(true)
        expect(getByTestId('Checkbox')).toBeTruthy();
    })
});
