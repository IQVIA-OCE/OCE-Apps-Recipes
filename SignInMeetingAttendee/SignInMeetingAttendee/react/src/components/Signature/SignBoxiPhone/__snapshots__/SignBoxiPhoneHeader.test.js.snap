// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signature Box Header for iPhone SignHeader Component with meal option checkbox for iPhone should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "marginTop": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 14,
              "letterSpacing": 0.25,
              "lineHeight": 20,
              "marginVertical": 2,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ],
          Array [
            Object {
              "fontSize": 14,
              "letterSpacing": 0.25,
              "lineHeight": 20,
              "marginVertical": 2,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ],
        ],
      ]
    }
  >
    Meal?
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLiveRegion="polite"
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessibilityTraits="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-only"
      style={Object {}}
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "borderRadius": 18,
                "paddingHorizontal": 6,
              },
              Object {
                "height": 34,
                "paddingVertical": 4,
                "width": 34,
              },
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Array [
                Object {
                  "color": "#0768FD",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ï„²
          </Text>
          <View
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "#0768FD",
                  "borderWidth": 0,
                  "height": 15,
                  "width": 15,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "borderRadius": 0,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Signature Box Header for iPhone SignHeader Component with picklist selectbox for iPhone should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "marginTop": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 14,
              "letterSpacing": 0.25,
              "lineHeight": 20,
              "marginVertical": 2,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ],
          Array [
            Object {
              "fontSize": 14,
              "letterSpacing": 0.25,
              "lineHeight": 20,
              "marginVertical": 2,
            },
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ],
        ],
      ]
    }
  >
    Meal?
  </Text>
  <View>
    <View
      style={
        Array [
          Object {
            "width": 160,
          },
          Object {
            "borderColor": "#C0C0C0",
            "borderWidth": 1,
            "marginLeft": 10,
          },
        ]
      }
    >
      <View>
        <View
          collapsable={false}
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "backgroundColor": undefined,
                  "borderColor": "#d9d9d9",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "elevation": 0,
                  "height": 40,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "paddingLeft": 8,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 14,
                    "left": 14,
                    "right": 14,
                    "top": 14,
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": "400",
                      },
                    ]
                  }
                >
                  Dinner
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType="button"
              accessibilityRole="button"
              accessibilityStates={Array []}
              accessibilityTraits="button"
              accessible={true}
              centered={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onLayout={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="box-only"
              style={Object {}}
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "margin": 6,
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 18,
                        "height": 36,
                        "width": 36,
                      },
                      undefined,
                      Object {
                        "margin": 0,
                      },
                    ],
                  ]
                }
              >
                <View>
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "transform": Array [
                              Object {
                                "scaleX": 1,
                              },
                            ],
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ],
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï…€
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "borderRadius": 0,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
